src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php

```diff
@@ -37,6 +37,36 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
     /**
      * {@inheritDoc}
      *
+     * The ORM implementation for preparing the object
+     */
+    public function prepareObject(TypeInterface $type, $parent = null)
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent){
+            throw new RuntimeException('You need a parent to create new objects');
+        }
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+        $metaDataParent = $this->om->getClassMetaData(get_class($parent));
```

Comment:
i would use Doctrine\Common\Util\ClassUtils::getRealClass for this, as you do not really need the metadata.